lang/ruby/test/test_protocol.rb

```diff
@@ -196,4 +219,12 @@ def test_namespaces
       assert_equal type.namespace, 'com.acme'
     end
   end
+
+  def test_support_for_doc_attribute
+    original = Protocol.parse(EXAMPLES.last.protocol_string)
+    # Testing doc support at the protocol level
+    assert_equal 'documentation', original.doc
+    # Testing doc support at the messages level
+    assert_equal 'documentation', original.messages['hello'].doc
```

Comment:
This is kind of nitpicky, but I'd probably use different strings like `'protocol_doc'` and `'message_doc'` for the two different doc values. Otherwise, technically you could make a mistake in the implementation and pull the doc from the same place for both assertions.